smriprep/workflows/outputs.py

```diff
@@ -655,12 +932,104 @@ def init_anat_derivatives_wf(
     return workflow
 
 
+def init_template_iterator_wf(*, spaces, name="template_iterator_wf"):
```

Comment:
I like this *a lot*. I feel like what we're going to want one of these sitting in the subject level that then gets passed as inputs to the baseline derivatives (registrations only), reporting and full derivative workflows.

Need to spend some time to work out what that would look like, in a separate PR. For now we spin one of these up inside the reporting and full derivatives workflows.